zifter/src/Zifter/Zift/Types.hs

```diff
@@ -108,8 +106,8 @@ data ZiftResult a
     deriving (Show, Eq, Generic)
 
 instance Validity a => Validity (ZiftResult a) where
-    isValid (ZiftSuccess a) = isValid a
-    isValid _ = True
+    validate (ZiftSuccess a) = validate a
```

Comment:
This is not semantics preserving. You'll have to take a very close look at the versioun bounds on validity.